float64 xgain
float64 ygain
float64 zgain
